zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -620,44 +615,7 @@ public boolean run(String paragraphId) {
   public boolean run(String paragraphId, boolean blocking) {
     Paragraph p = getParagraph(paragraphId);
     p.setListener(jobListenerFactory.getParagraphJobListener(this));
-    
```

Comment:
Move this into Paragraph.java. Paragraph has sufficient info to run itself instead of doing it in Note.